### PR TITLE
fix(markets): migrate currency symbol from $ to Ƀ [Stack 1/2]

### PR DESCRIPTION
**Pull request by slkzgm**
## Summary

- **Migrate all currency displays from `$` to `Ƀ` (Babylon Points symbol)** across market components
- Replace `DollarSign` icons with `Coins` icons for visual consistency
- Ensure ticker names (e.g., `$BTC`, `$ETH`) retain the `$` prefix as intended

> ⚠️ **Stacked PR (1/2)** - This is the first PR in a stack. Merge this before [Stack 2/2].

## Type

- [ ] Feature
- [x] Bug fix
- [ ] Refactor / cleanup (no behavior change)
- [ ] Performance
- [ ] Infra / ops
- [ ] Docs
- [ ] Contracts / on-chain
- [ ] Other: …

## Context / Links

- Part of BAB-69 market improvements
- Currency symbol migration from USD (`$`) to Babylon Points (`Ƀ`)

## Scope (keep it focused)

- [x] This PR is focused on a single change/theme (not a catch‑all)
- [x] Drive‑by refactors are excluded or split into a separate PR
- [x] Non‑goals / follow‑ups are listed below (with links)

**Areas touched**
- [x] Web UI (`apps/web`)
- [ ] Web API routes / SSE / A2A (`apps/web`)
- [ ] CLI (`apps/cli`)
- [ ] Docs site (`apps/docs`) / vendor docs (`docs/vendors/*`)
- [ ] Domain / game engine (`packages/engine`, `packages/core/*`)
- [ ] Agents / runtime / A2A / MCP (`packages/agents`, `packages/a2a`, `packages/mcp`)
- [ ] API infra (`packages/api`)
- [ ] DB (`packages/db`)
- [ ] Contracts / on-chain (`packages/contracts`)
- [ ] Shared types/utils (`packages/shared`)
- [ ] Tests (`packages/testing`)
- [ ] Other: …

### Icon Changes
- `MarketOverviewPanel.tsx`: `DollarSign` → `Coins`
- `TradingProfile.tsx`: `DollarSign` → `Coins`

### Currency Formatting Updates
| File | Change |
|------|--------|
| `MarketOverviewPanel.tsx` | `formatVolume` uses `BABYLON_POINTS_SYMBOL` |
| `PerpPositionsList.tsx` | `formatPrice` + toast messages |
| `PredictionPositionsList.tsx` | `formatPrice` + toast messages |
| `TopMoversPanel.tsx` | `formatPrice` |
| `PnLShareModal.tsx` | Share text messages |
| `TradeConfirmationDialog.tsx` | `formatPrice` |
| `AssetTradesFeed.tsx` | `formatCurrency` |
| `CategoryPnLShareCard.tsx` | `formatCurrency` |
| `PortfolioPnLShareCard.tsx` | `formatCurrency` |
| `PortfolioPnLCard.tsx` | `formatCurrency` |
| `PerpTradingModal.tsx` | Toast messages |
| `LightweightChartBase.tsx` | `formatChartPrice` |
| `TradingProfile.tsx` | `formatCurrency` |
| `predictions/[id]/page.tsx` | Text: "$1 per share" → "Ƀ1 per share" |
| `PerpPriceChart.test.tsx` | Test expectations |

## Review guide

**Start here**
- `apps/web/src/components/markets/MarketOverviewPanel.tsx` - typical example of the pattern

**Risk**
- [x] Low
- [ ] Medium
- [ ] High

**Notes for reviewers**
- All changes follow the same pattern: import `BABYLON_POINTS_SYMBOL` from `@babylon/shared` and use it in format functions
- Ticker names like `$BTC` intentionally keep the `$` prefix (these are stock ticker symbols, not currency)
- The constant `BABYLON_POINTS_SYMBOL = "Ƀ"` is already defined in `@babylon/shared`

## Test plan

**Commands run**
- [x] `bun run check` (Biome format)
- [ ] `bun run typecheck`
- [ ] `bun run lint`
- [ ] `bun run build`
- [ ] `bun run test` (unit + integration)
- [ ] `bun run test:e2e` (if critical flows changed)
- [ ] `bun run contracts:test` (if contracts changed)
- [ ] `bun run db:check` (if DB schema/queries changed)

**Manual verification**
1. Navigate to `/markets` and verify all prices show `Ƀ` instead of `$`
2. Open a perp market detail page, verify chart prices show `Ƀ`
3. Open a prediction market, verify prices show `Ƀ`
4. Open a position and verify toast messages show `Ƀ`
5. Verify ticker names still show `$BTC`, `$ETH`, etc.

## Ops / Migration / Deployment

- [x] No deploy impact
- [ ] Requires env var updates (listed below + `.env.example` updated)
- [ ] Requires DB migration (`bun run db:migrate`) / backfill / seed
- [ ] Changes cron schedule or endpoints (`vercel.json`, `CRON_SECRET`, etc.)
- [ ] Rollout behind a flag / gradual rollout

## Breaking changes

- [x] None
- [ ] Yes (describe + migration guide below)

## Security / privacy

- [x] No security impact
- [ ] Needs extra review (describe)

### Option B: No visual changes

- Reason: This is a text/symbol replacement only. The UI layout and behavior remain identical. The only visible change is `$` → `Ƀ` in currency displays (not ticker names).

## Checklist (author)
- [x] Self-review done (diff + critical paths)
- [x] Base branch is correct (`feat/bab-69-market-volatility-simulation`)
- [x] Handlers remain thin and portable (validate → service → map errors)
- [x] Domain logic stays in packages (no Next/React/Elysia coupling in core)
- [ ] `.env.example` updated (if env changed) and variables documented above
- [ ] Docs updated (and `bun run docs:generate` if needed)
- [x] Tests added/updated for behavior changes (or rationale provided)
- [ ] Dependency changes are intentional (`bun.lock` updated)
- [ ] Code owners requested (auto via CODEOWNERS or manual)
- [x] **Screenshots/recordings section filled** (visual demo OR explanation why N/A)

## Summary by CodeRabbit

* **Style**
  * Switched currency displays and UI labels to use the Babylon Points symbol (Ƀ) and updated related icons (coin icons replacing dollar icons).
* **New Features**
  * Added unified, configurable currency formatting with compact K/M/B suffixes and sign-aware output.
* **Tests**
  * Added comprehensive tests covering currency, compact-formatting, and share-text PnL formatting.
* **Documentation**
  * Updated help and formatting text (e.g., "If you're right, you'll receive Ƀ1 per share.").

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

<h3>Greptile Summary</h3>

This PR successfully migrates all currency displays from USD (`$`) to Babylon Points (`Ƀ`) across the markets UI. The changes are focused and systematic:

- Replaces hardcoded `$` and `Intl.NumberFormat` USD formatters with `BABYLON_POINTS_SYMBOL` constant from `@babylon/shared`
- Updates icons from `DollarSign` to `Coins` in two components for visual consistency
- Correctly preserves `$` prefix for ticker symbols (e.g., `$BTC`, `$ETH`) as intended
- Updates test expectations to reflect new symbol
- Covers all market-related components: positions lists, trading modals, charts, PnL cards, and share functionality

The implementation is clean and maintains backward compatibility for ticker symbol display. All changes follow the same pattern: import the constant and use string interpolation for formatting.

<h3>Confidence Score: 5/5</h3>

- Safe to merge - straightforward symbol replacement with consistent pattern
- All changes follow a uniform pattern of replacing hardcoded `$` with `BABYLON_POINTS_SYMBOL` constant. The implementation is consistent across all 15 files, correctly distinguishes between currency symbols (changed) and ticker prefixes (preserved), and includes updated test expectations.
- No files require special attention

<h3>Important Files Changed</h3>

| Filename | Overview |
|----------|----------|
| apps/web/src/components/markets/MarketOverviewPanel.tsx | Replaced `$` with `BABYLON_POINTS_SYMBOL` in volume formatting and changed `DollarSign` icon to `Coins` |
| apps/web/src/components/markets/PerpPositionsList.tsx | Replaced `Intl.NumberFormat` with `BABYLON_POINTS_SYMBOL` formatting, correctly preserves `$` prefix for ticker display |
| apps/web/src/components/markets/TradeConfirmationDialog.tsx | Updated price formatting to use `BABYLON_POINTS_SYMBOL`, correctly preserves `$` prefix for ticker names in descriptions |
| apps/web/src/components/markets/AssetTradesFeed.tsx | Replaced `Intl.NumberFormat` USD currency formatter with simple `BABYLON_POINTS_SYMBOL` concatenation |
| apps/web/src/components/charts/LightweightChartBase.tsx | Replaced hardcoded `$` with `BABYLON_POINTS_SYMBOL` in chart price formatting |

</details>

<h3>Sequence Diagram</h3>

```mermaid
sequenceDiagram
    participant User
    participant Component as UI Component
    participant Shared as @babylon/shared
    participant Format as Formatter Function
    
    Note over User,Format: Currency Symbol Migration Flow
    
    User->>Component: Views market data
    Component->>Shared: Import BABYLON_POINTS_SYMBOL
    Shared-->>Component: Returns 'Ƀ' constant
    
    Component->>Format: Call formatPrice(123.45)
    Format->>Shared: Use BABYLON_POINTS_SYMBOL
    Shared-->>Format: 'Ƀ'
    Format-->>Component: 'Ƀ123.45'
    Component-->>User: Display 'Ƀ123.45'
    
    Note over Component,Format: Ticker symbols preserved
    Component->>User: Display '$BTC' (unchanged)
    
    Note over User,Format: Toast Notifications
    User->>Component: Execute trade
    Component->>Format: Format PnL with BABYLON_POINTS_SYMBOL
    Format-->>Component: 'Ƀ50.00'
    Component->>User: Toast: '+Ƀ50.00 PnL'
```

### CODE REVIEW

File: apps/web/src/components/markets/PnLShareModal.tsx [hunk 1]
```diff
@@ -152,10 +157,10 @@ export function PnLShareModal({
       'Join me in Babylon, a real-time simulation where humans and AI agents battle across prediction markets, form alliances, and shape outcomes—together.';
 
     if (type === 'portfolio' && portfolioData) {
-      return `My Babylon P&L is ${portfolioData.totalPnL >= 0 ? '+' : ''}$${Math.abs(portfolioData.totalPnL).toFixed(2)}. Trading narratives, sharing the upside.\n\n${standardMessage}\n\n${link}`;
+      return `My Babylon P&L is ${portfolioData.totalPnL >= 0 ? '+' : ''}${BABYLON_POINTS_SYMBOL}${Math.abs(portfolioData.totalPnL).toFixed(2)}. Trading narratives, sharing the upside.\n\n${standardMessage}\n\n${link}`;
```
[the remark this answers is not in the document]
Comment (slkzgm) [reply apps/web/src/components/markets/PnLShareModal.tsx]: **Addressed in commit `d77a4455f`**

Fixed by extracting the sign calculation into a variable:
```typescript
const sign = portfolioData.totalPnL >= 0 ? '+' : '-';
return `...${sign}${BABYLON_POINTS_SYMBOL}${Math.abs(...)}`;
```

Now produces:
- Positive: `+Ƀ100.00`
- Negative: `-Ƀ100.00`

Added unit tests in `PnLShareModal.test.tsx` to verify the fix.

File: apps/web/src/components/markets/PnLShareModal.tsx [hunk 2]
```diff
@@ -152,10 +157,10 @@ export function PnLShareModal({
...
     }
     if (type === 'category' && categoryData) {
-      return `My ${categoryLabel} P&L on Babylon is ${categoryData.unrealizedPnL >= 0 ? '+' : ''}$${Math.abs(categoryData.unrealizedPnL).toFixed(2)}. Trading narratives, sharing the upside.\n\n${standardMessage}\n\n${link}`;
+      return `My ${categoryLabel} P&L on Babylon is ${categoryData.unrealizedPnL >= 0 ? '+' : ''}${BABYLON_POINTS_SYMBOL}${Math.abs(categoryData.unrealizedPnL).toFixed(2)}. Trading narratives, sharing the upside.\n\n${standardMessage}\n\n${link}`;
```
[the remark this answers is not in the document]
Comment (slkzgm) [reply apps/web/src/components/markets/PnLShareModal.tsx]: **Addressed in commit `d77a4455f`**

Same fix applied for category P&L:
```typescript
const sign = categoryData.unrealizedPnL >= 0 ? '+' : '-';
```

Both portfolio and category P&L now correctly display the minus sign for losses.

File: apps/web/src/components/markets/AssetTradesFeed.tsx [hunk 3]
```diff
@@ -347,12 +347,7 @@ export function AssetTradesFeed({
...
   const formatCurrency = (value: string | number) => {
     const num = typeof value === 'string' ? Number.parseFloat(value) : value;
-    return new Intl.NumberFormat('en-US', {
-      style: 'currency',
-      currency: 'USD',
-      minimumFractionDigits: 2,
-      maximumFractionDigits: 2,
-    }).format(num);
+    return `${BABYLON_POINTS_SYMBOL}${num.toFixed(2)}`;
   };
```
[the remark this answers is not in the document]
Comment (slkzgm) [reply apps/web/src/components/markets/AssetTradesFeed.tsx:356]: **Addressed in commit `695fceb40`**

Refactored to use shared `formatCurrency` from `@babylon/shared`:
```typescript
import { formatCurrency as formatCurrencyShared } from '@babylon/shared';

const formatCurrency = (value: string | number) => {
  const num = typeof value === 'string' ? Number.parseFloat(value) : value;
  return formatCurrencyShared(num);
};
```

Kept a local wrapper to handle string input for backwards compatibility with existing call sites.

File: apps/web/src/components/markets/PerpPositionsList.tsx [hunk 4]
```diff
@@ -140,12 +144,7 @@ export function PerpPositionsList({
...
 
   const formatPrice = (price: number) => {
-    return new Intl.NumberFormat('en-US', {
-      style: 'currency',
-      currency: 'USD',
-      minimumFractionDigits: 2,
-      maximumFractionDigits: 2,
-    }).format(price);
+    return `${BABYLON_POINTS_SYMBOL}${price.toFixed(2)}`;
   };
```
[the remark this answers is not in the document]
Comment (slkzgm) [reply apps/web/src/components/markets/PerpPositionsList.tsx]: **Addressed in commit `695fceb40`**

Replaced local `formatPrice` implementation with shared `formatCurrency` from `@babylon/shared`:
```typescript
import { formatCurrency } from '@babylon/shared';

const formatPrice = (price: number) => formatCurrency(price);
```

File: apps/web/src/components/markets/PredictionPositionsList.tsx [hunk 5]
```diff
@@ -143,7 +143,7 @@ export function PredictionPositionsList({
     }
   };
 
-  const formatPrice = (price: number) => `$${price.toFixed(3)}`;
+  const formatPrice = (price: number) => `${BABYLON_POINTS_SYMBOL}${price.toFixed(3)}`;
```
[the remark this answers is not in the document]
Comment (slkzgm) [reply apps/web/src/components/markets/PredictionPositionsList.tsx]: **Addressed in commit `695fceb40`**

Replaced local implementation with shared `formatCurrency` from `@babylon/shared`:
```typescript
import { formatCurrency } from '@babylon/shared';

const formatPrice = (price: number) => formatCurrency(price, 3);
```

Using decimals=3 to match the original behavior for prediction share prices.

File: apps/web/src/components/markets/TopMoversPanel.tsx [hunk 6]
```diff
@@ -35,7 +35,7 @@ export function TopMoversPanel({ onMarketClick }: TopMoversPanelProps) {
   const { topGainers, topLosers, loading } = usePerpTopMovers(4);
   usePerpMarketsPolling(30000); // Enable 30s polling
 
-  const formatPrice = (p: number) => `$${p.toFixed(2)}`;
+  const formatPrice = (p: number) => `${BABYLON_POINTS_SYMBOL}${p.toFixed(2)}`;
```
[the remark this answers is not in the document]
Comment (slkzgm) [reply apps/web/src/components/markets/TopMoversPanel.tsx]: **Addressed in commit `695fceb40`**

Replaced local implementation with shared `formatCurrency` from `@babylon/shared`:
```typescript
import { formatCurrency } from '@babylon/shared';

const formatPrice = (p: number) => formatCurrency(p);
```

File: apps/web/src/components/markets/TradeConfirmationDialog.tsx [hunk 7]
```diff
@@ -155,12 +155,7 @@ export function TradeConfirmationDialog({
...
 
   const formatPrice = (price: number) => {
-    return new Intl.NumberFormat('en-US', {
-      style: 'currency',
-      currency: 'USD',
-      minimumFractionDigits: 2,
-      maximumFractionDigits: 2,
-    }).format(price);
+    return `${BABYLON_POINTS_SYMBOL}${price.toFixed(2)}`;
   };
```
[the remark this answers is not in the document]
Comment (slkzgm) [reply apps/web/src/components/markets/TradeConfirmationDialog.tsx]: **Addressed in commit `695fceb40`**

Replaced local implementation with shared `formatCurrency` from `@babylon/shared`:
```typescript
import { formatCurrency } from '@babylon/shared';

const formatPrice = (price: number) => formatCurrency(price);
```

File: apps/web/src/components/profile/TradingProfile.tsx [hunk 8]
```diff
@@ -312,11 +312,13 @@ export function TradingProfile({
...
     const abs = Math.abs(value);
-    if (abs >= 1000000) return `$${(value / 1000000).toFixed(2)}M`;
-    if (abs >= 1000) return `$${(value / 1000).toFixed(2)}K`;
-    return `$${value.toFixed(2)}`;
+    if (abs >= 1000000)
+      return `${BABYLON_POINTS_SYMBOL}${(value / 1000000).toFixed(2)}M`;
+    if (abs >= 1000)
+      return `${BABYLON_POINTS_SYMBOL}${(value / 1000).toFixed(2)}K`;
+    return `${BABYLON_POINTS_SYMBOL}${value.toFixed(2)}`;
   };
```
[the remark this answers is not in the document]
Comment (slkzgm) [reply apps/web/src/components/profile/TradingProfile.tsx]: **Addressed in commit `695fceb40`**

Replaced local implementation with shared `formatCompactCurrency` from `@babylon/shared`:
```typescript
import { formatCompactCurrency } from '@babylon/shared';

const formatCurrency = (value: number) => formatCompactCurrency(value);
```

The shared utility handles:
- Non-finite value guard (`Ƀ0.00` fallback)
- Compact formatting (K/M/B suffixes)
- Negative value handling with correct sign placement

**Comment by slkzgm** [comment]
## Review Findings Summary

All actionable findings from the CodeRabbit review have been addressed. Below is the complete breakdown:

---

### ✅ Addressed Findings

| # | File | Issue | Resolution | Commit |
|---|------|-------|------------|--------|
| 1 | `PnLShareModal.tsx` | **Critical**: Negative portfolio P&L values lose minus sign | Fixed by using explicit `'+' : '-'` ternary | `d77a4455f` |
| 2 | `PnLShareModal.tsx` | **Critical**: Negative category P&L values lose minus sign | Same fix applied | `d77a4455f` |
| 3 | `AssetTradesFeed.tsx` | Local `formatCurrency` duplicates shared utility | Refactored to use `formatCurrency` from `@babylon/shared` | `695fceb40` |
| 4 | `PerpPositionsList.tsx` | Local `formatPrice` duplicates shared utility | Refactored to use `formatCurrency` from `@babylon/shared` | `695fceb40` |
| 5 | `PredictionPositionsList.tsx` | Local `formatPrice` duplicates shared utility | Refactored to use `formatCurrency(price, 3)` for 3 decimals | `695fceb40` |
| 6 | `TopMoversPanel.tsx` | Local `formatPrice` duplicates shared utility | Refactored to use `formatCurrency` from `@babylon/shared` | `695fceb40` |
| 7 | `TradeConfirmationDialog.tsx` | Local `formatPrice` duplicates shared utility | Refactored to use `formatCurrency` from `@babylon/shared` | `695fceb40` |
| 8 | `TradingProfile.tsx` | Local `formatCurrency` with compact (K/M) duplicates logic | Refactored to use new `formatCompactCurrency` from `@babylon/shared` | `695fceb40` |
| 9 | `predictions/[id]/page.tsx` | Label inconsistency: "Amount (PTS)" vs symbol "Ƀ" | Updated label to "Amount (Ƀ)" | `0ef11c58` |

---

### 🔧 Additional Improvements

As part of addressing the DRY findings, the following utilities were added to `@babylon/shared`:

- **`formatCurrency(amount, options)`**: Enhanced with optional `useThousandsSeparator` flag and proper negative number handling
- **`formatCompactCurrency(value, decimals, threshold)`**: New utility for K/M/B suffix formatting

Unit tests added in `packages/testing/unit/shared/format.test.ts` to cover:
- Standard formatting with Ƀ symbol
- Thousands separator behavior
- Negative number sign placement
- Compact currency formatting (K/M/B)

---

### ❌ Skipped Findings

None. All 8 inline comments and 1 outside-diff comment were valid and have been addressed.

File: apps/web/src/components/markets/PerpPositionsList.tsx [hunk 9]
```diff
@@ -129,7 +134,7 @@ export function PerpPositionsList({
           : 0;
 
     toast.success('Position closed!', {
-      description: `${pendingClose.position.ticker}: ${pnl >= 0 ? '+' : ''}$${pnl.toFixed(2)} PnL`,
+      description: `${pendingClose.position.ticker}: ${pnl >= 0 ? '+' : ''}${BABYLON_POINTS_SYMBOL}${pnl.toFixed(2)} PnL`,
```
[the remark this answers is not in the document]
Comment (slkzgm) [reply apps/web/src/components/markets/PerpPositionsList.tsx]: **Addressed in commit `9765c2fac`**

Fixed by extracting the sign and using `Math.abs()`:
```typescript
const pnlSign = pnl >= 0 ? '+' : '-';
toast.success('Position closed!', {
  description: `${pendingClose.position.ticker}: ${pnlSign}${BABYLON_POINTS_SYMBOL}${Math.abs(pnl).toFixed(2)} PnL`,
});
```

Now correctly produces:
- Positive: `$BTC: +Ƀ100.00 PnL`
- Negative: `$BTC: -Ƀ100.00 PnL`

File: apps/web/src/components/markets/PredictionPositionsList.tsx [hunk 10]
```diff
@@ -124,7 +129,7 @@ export function PredictionPositionsList({
       const data: SellSharesSuccessResponse = await response.json();
       const pnl = data.pnl;
       toast.success('Shares sold!', {
-        description: `Sold ${position.shares.toFixed(2)} ${position.side} shares for ${pnl >= 0 ? '+' : ''}$${pnl.toFixed(2)} PnL`,
+        description: `Sold ${position.shares.toFixed(2)} ${position.side} shares for ${pnl >= 0 ? '+' : ''}${BABYLON_POINTS_SYMBOL}${pnl.toFixed(2)} PnL`,
```
[the remark this answers is not in the document]
Comment (slkzgm) [reply apps/web/src/components/markets/PredictionPositionsList.tsx]: **Addressed in commit `9765c2fac`**

Same fix applied:
```typescript
const pnlSign = pnl >= 0 ? '+' : '-';
toast.success('Shares sold!', {
  description: `Sold ... for ${pnlSign}${BABYLON_POINTS_SYMBOL}${Math.abs(pnl).toFixed(2)} PnL`,
});
```

Sign is now correctly placed before the symbol for negative values.

File: packages/shared/src/utils/format.ts [hunk 11]
```diff
@@ -157,24 +157,102 @@ export function formatCompactNumber(num: number): string {
...
+  }
+  if (abs >= 1_000_000) {
+    return `${sign}${BABYLON_POINTS_SYMBOL}${(abs / 1_000_000).toFixed(decimals)}M`;
+  }
+  if (abs >= 1_000) {
+    return `${sign}${BABYLON_POINTS_SYMBOL}${(abs / 1_000).toFixed(decimals)}K`;
+  }
+
+  return `${sign}${BABYLON_POINTS_SYMBOL}${abs.toFixed(decimals)}`;
 }
```
[the remark this answers is not in the document]
Comment (slkzgm) [reply packages/shared/src/utils/format.ts:256]: **Addressed in commit `9765c2fac`**

Fixed by using `toFixed(decimals)` on 0 instead of string concatenation:
```typescript
// Before (produced 'Ƀ0.' when decimals=0)
return \`\${BABYLON_POINTS_SYMBOL}0.\${'0'.repeat(decimals)}\`;

// After (correctly produces 'Ƀ0' when decimals=0)
return \`\${BABYLON_POINTS_SYMBOL}\${(0).toFixed(decimals)}\`;
```

Added unit test to verify the edge case:
```typescript
it('should handle non-finite values with decimals=0 (no trailing dot)', () => {
  expect(formatCompactCurrency(NaN, 0)).toBe('Ƀ0');
  expect(formatCompactCurrency(Infinity, 0)).toBe('Ƀ0');
});
```

File: apps/web/src/components/markets/PnLShareModal.tsx [hunk 12]
```diff
@@ -152,10 +157,12 @@ export function PnLShareModal({
...
 
     if (type === 'portfolio' && portfolioData) {
-      return `My Babylon P&L is ${portfolioData.totalPnL >= 0 ? '+' : ''}$${Math.abs(portfolioData.totalPnL).toFixed(2)}. Trading narratives, sharing the upside.\n\n${standardMessage}\n\n${link}`;
+      const sign = portfolioData.totalPnL >= 0 ? '+' : '-';
+      return `My Babylon P&L is ${sign}${BABYLON_POINTS_SYMBOL}${Math.abs(portfolioData.totalPnL).toFixed(2)}. Trading narratives, sharing the upside.\n\n${standardMessage}\n\n${link}`;
     }
     if (type === 'category' && categoryData) {
-      return `My ${categoryLabel} P&L on Babylon is ${categoryData.unrealizedPnL >= 0 ? '+' : ''}$${Math.abs(categoryData.unrealizedPnL).toFixed(2)}. Trading narratives, sharing the upside.\n\n${standardMessage}\n\n${link}`;
+      const sign = categoryData.unrealizedPnL >= 0 ? '+' : '-';
+      return `My ${categoryLabel} P&L on Babylon is ${sign}${BABYLON_POINTS_SYMBOL}${Math.abs(categoryData.unrealizedPnL).toFixed(2)}. Trading narratives, sharing the upside.\n\n${standardMessage}\n\n${link}`;
```
[the remark this answers is not in the document]
Comment (slkzgm) [reply apps/web/src/components/markets/PnLShareModal.tsx:165]: **Skipped - Not applicable**

The current implementation is correct and the sign handling is already properly implemented with the explicit `sign` variable. 

Using shared `formatCurrency` here would require additional string manipulation to add the sign prefix since the share message format requires `+` for positive values (which `formatCurrency` doesn't add). The current inline approach is cleaner for this specific use case.

**Comment by slkzgm** [comment]
## Review Findings - Round 2 Summary

All new findings from the latest CodeRabbit review have been processed. Commit: `9765c2fac`

---

### ✅ Addressed Findings

| # | File | Severity | Issue | Resolution |
|---|------|----------|-------|------------|
| 1 | `PerpPositionsList.tsx:137` | 🟠 Major | Toast shows `Ƀ-100` instead of `-Ƀ100` for negative PnL | Fixed with explicit `pnlSign` variable and `Math.abs()` |
| 2 | `PredictionPositionsList.tsx:132` | 🟠 Major | Same sign placement issue in sell shares toast | Same fix applied |
| 3 | `format.ts:237` | 🟡 Minor | `formatCompactCurrency(NaN, 0)` produces `Ƀ0.` (trailing dot) | Fixed by using `(0).toFixed(decimals)` instead of string concat |

### ⏭️ Skipped Findings

| # | File | Severity | Issue | Reason |
|---|------|----------|-------|--------|
| 1 | `PnLShareModal.tsx:165` | 🔵 Nitpick | Suggests using shared `formatCurrency` | Current implementation is correct. Using shared utility would require additional manipulation since share messages need `+` prefix for positive values (which `formatCurrency` doesn't provide). |

---

### 🧪 Tests Added

- `format.test.ts`: Added test case for `formatCompactCurrency` with `decimals=0` on non-finite values to prevent regression
